MSAL/test/automation/tests/MSALBaseUITest.m

```diff
@@ -479,12 +528,31 @@ - (MSIDAutomationAccountsResult *)automationAccountsResult
 
 - (NSDictionary *)automationResultDictionary
 {
-    XCUIElement *resultTextView = self.testApp.textViews[@"resultInfo"];
-    [self waitForElement:resultTextView];
+    int timeout = 10;
+    __auto_type resultPipelineExpectation = [[XCTestExpectation alloc] initWithDescription:@"Wait for result pipeline."];
+    
+    // Wait till file appears.
+    int i = 0;
+    while (i < timeout)
+    {
+        if ([NSFileManager.defaultManager fileExistsAtPath:[MSIDAutomationActionConstants resultPipelinePath]])
```

Comment:
Is there a way to observe file changes?

Reply:
+1 this sounds very random, and maybe observe the size of a file?

Reply:
We can observe changes in the directory, but there isn't high level api, we will need to implement the code. I prefer to keep the current solution since it proved to be stable and simple.